### PR TITLE
Skynet Substrate - Milestone 1

### PR DESCRIPTION
**Pull request by dghelm**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/489
Amendment: https://github.com/w3f/Grants-Program/pull/845

### CODE REVIEW

**Comment by Noc2** [comment]
Thanks for the delivery. We will look into it as soon as possible.

File: deliveries/skynet-substrate-integration-1.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).**
+
+- **Application Document:** https://github.com/w3f/Grants-Program/pull/845
```
Comment (alxs) [inline deliveries/skynet-substrate-integration-1.md:5]:
```suggestion
- **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/skynet-substrate-integration.md
```

**Review by alxs — CHANGES_REQUESTED** [review]

File: deliveries/skynet-substrate-integration-1.md (hunk 1 above)
Comment (dghelm) [reply deliveries/skynet-substrate-integration-1.md]: Whoops -- thanks.

**Comment by dghelm** [comment]
> Thanks again for the delivery. I can confirm that the basic example works and generally it looks pretty good. However (in case I’m not missing something), most of it was implemented in rust, but not actually as a pallet (see deliverables Off-Chain Worker Pallet). Also it would be nice to have a tutorial or “documentation that explains how a user can use the SDK in a pallet for basic off-chain functionality.” to make it easier for others to actually use this for their project. You can find my evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/skynet-substrate-integration_Noc2.md).

Thanks for the feedback and evaluation!

I'll coordinate on the documentation and warning message with the team. Regarding the front-end issues, siasky.net is our public portal that is struggling under the weight of anonymous traffic. It might be better to use this community portal while we roll out our new portals and fixes: https://000a09kgtkertvla0vdksg42rihmgg1armrfa245krprc67nrap5pm0.fileportal.org/#/

Bigger picture question:
I amended the language of the proposal to specifically mention that we're building an SDK for use by OCW Pallets. I hope I'm not misusing the Substrate jargon, but as we dove into the ecosystem, a _specific_ runtime behavior on Skynet seemed much less interesting than allowing developers to incorporate off-chain storage into any off-chain worker (which I understood to be them creating their own ocw pallet like in our runtime example). Am I using this language correctly? Do you have suggestions on either how we can clarify what the SDK does in our proposal, or how we might augment our repos in order to get a successful evaluation and make this work more interesting to other teams?

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the clarification. Yes, you are using the language correctly. I overlooked the amendment during the evaluation (even though I used it for my notes). Sorry for this. My only outstanding request would be to add some documentation or tutorial to make it easier for devs to leverage your work. If you want you could even publish this as a medium post or something similar. Let me know if you have any questions.

**Comment by dghelm** [comment]
> Thanks for the clarification. Yes, you are using the language correctly. I overlooked the amendment during the evaluation (even though I used it for my notes). Sorry for this. My only outstanding request would be to add some documentation or tutorial to make it easier for devs to leverage your work. If you want you could even publish this as a medium post or something similar. Let me know if you have any questions.

Awesome!

The warning message should [now be fixed](https://github.com/SkynetLabs/skynet-substrate/commit/9f768c8f0f1ffd7f875b7177bbffde8e386f143d).

Our main developer documentation is at https://docs.skynetlabs.com/ and I've written an initial draft for what we will devote a page to in the "integrations" section, once we're ready to announce this work. The aim is less tutorial, and more a run-down of the Skynet-specific interactions so a developer can easily see the moving pieces of the off-chain worker. We'll also link back to this in the repo once published. I'd love feedback as someone who is coming from Substrate ecosystem and not the Skynet ecosystem!
https://skynetlabs.notion.site/Skynet-Substrate-SDK-5d0e02a25a43419ba39f30da071517dd

In addition to this, we're planning to do some promotion around the grant milestone + SDK release, which will include a blog post with a high-level explanation of the SDK, what our example repo is doing, why we think it's important, and how to get started (likely here we'll also link to substrate docs for devs that have no experience with substrate).

**Review by Noc2 — APPROVED** [review]
Thanks a lot for the update and the great article. That’s exactly what I meant. I’m happy to tell you that the milestone is a pass and I will forward your invoice internally. Regarding a potential announcement feel free to reach out to grantspr@web3.foundation

**Comment by RouvenP** [comment]
hi @dghelm we transferred the payment today. thanks!

**Pull request merged**